### PR TITLE
Reimagine Sharing: Export audio clips

### PR DESCRIPTION
**Pull request by bjtitus**
| 📘 Part of: #1910  |
|:---:|

Adds exporting of audio only clips:

https://github.com/user-attachments/assets/0b02730b-b0e2-4388-b166-dac687c44a07

### Known Issues
* The current "clip.m4a" file name is a placeholder. File names cannot be dynamic with the `TransferRepresentation` APIs until iOS 17 😮‍💨 so I'll be changing the sharing code back to `NSItemProvider` in a separate PR.
* The title and description should change when swiping to the Audio style but I need to do a bit of refactoring to prevent a jump with the `TabView`.

## To test

> [!NOTE]
> Enable the `newSharing` feature flag to test this

* Play an episode
* Tap on the "Share" button in the action shelf
* Tap on "Clip" from the share modal
* Create a clip and swipe over the the audio symbol
* Tap "Clip"
* Tap the "More" share option and share or save to a file
* Verify that an `m4a` audio file is shared
* Verify that the `m4a` file is playable with correct audio

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/57">7.71</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Sharing/SharingModal.swift [hunk 1]
```diff
@@ -162,19 +162,26 @@ extension SharingModal.Option {
...
             guard let fileURL = progress.fileURL else {
-                return ShareImageView(info: imageInfo, style: style, angle: .constant(0)).itemProvider()
+                return [ShareImageView(info: imageInfo, style: style, angle: .constant(0)).itemProvider()]
             }
             if #available(iOS 16.0, *) {
-                return NSItemProvider(contentsOf: fileURL, contentType: UTType(filenameExtension: fileURL.pathExtension) ?? .mpeg4Movie)
+                let mediaItemProvider = NSItemProvider()
+                mediaItemProvider.suggestedName = "\(imageInfo.title) - \(imageInfo.description)"
+                mediaItemProvider.registerDataRepresentation(for: UTType(filenameExtension: fileURL.pathExtension) ?? .mpeg4Movie) { completion in
+                    completion(try! Data(contentsOf: progress.fileURL!), nil)
```
Comment (danielebogo) [inline podcasts/Sharing/SharingModal.swift]: Quick one: Can we avoid the force try here?

File: podcasts/Sharing/SharingModal.swift [hunk 2]
```diff
@@ -162,19 +162,26 @@ extension SharingModal.Option {
...
+                    return nil
+                }
+                let urlItemProvider = NSItemProvider(object: URL(string: shareURL)! as NSURL)
+                return [mediaItemProvider]//, urlItemProvider]
             } else {
-                return NSItemProvider(contentsOf: fileURL)!
+                return [NSItemProvider(contentsOf: fileURL)!]
```
Comment (danielebogo) [inline podcasts/Sharing/SharingModal.swift]: Same here

**Review by danielebogo — CHANGES_REQUESTED** [review]
Hey @bjtitus , following the steps, the feature works. I was able to export the audio, save it and listen to the clip 👍 

I left just few comments, but I have some questions:
• The first clip image goes a bit under the pagination dots. Is it expected?
• While exporting, I can still swipe between the different types. Is it correct? Because, if I export the first type, while it's exporting, I can swipe but the animation starts to be choppy.
• If I re-edit the clip while exporting, and tap on clip again, I don't see the progress anymore

**Comment by bjtitus** [comment]
> The first clip image goes a bit under the pagination dots. Is it expected?

Just fixed this in https://github.com/Automattic/pocket-casts-ios/pull/2045

> I can still swipe between the different types

Only the selected style should be shown. I've fixed that in [5173479](https://github.com/Automattic/pocket-casts-ios/pull/2044/commits/51734799fc5ad55def66fce34c5e3e902719c75b)

> If I re-edit the clip while exporting, and tap on clip again, I don't see the progress anymore

The cancellation got messed up when I moved this to the `task` ViewModifier. Could you retry now? The `VideoExporter` still needs a little extra work on cancellations but I'll submit that separately.

**Review by danielebogo — APPROVED** [review]
It looks good. Thanks for fixing it after the feedback!

**Pull request merged**